PyInstaller 4.3 bootloader fails to compile on CentOS 7

PyInstaller [4.3 has recently been released](https://github.com/pyinstaller/pyinstaller/blob/e20e74c03768d432d48665b8ef1e02511b16e4be/doc/CHANGES.rst#43-2021-04-16), however, changes made to the bootloader in this release result in compilation failures on CentOS 7 (and/or gcc 4.8.5) as follows:

```
Setting top to                           : /gridsync/build/pyinstaller/bootloader 
Setting out to                           : /gridsync/build/pyinstaller/bootloader/build 
Python Version                           : 3.9.2 (default, Mar  1 2021, 07:19:40) [GCC 4.8.5 20150623 (Red Hat 4.8.5-44)] 
Checking for 'gcc' (C compiler)          : /usr/bin/gcc 
System                                   : Assuming cross-compilation for Linux 
Checking size of pointer                 : 8 
Platform                                 : Linux-64bit detected based on compiler 
Checking for compiler flags -m64         : yes 
Checking for linker flags -m64           : yes 
Checking for library dl                  : yes 
Checking for library pthread             : yes 
Checking for library m                   : yes 
Checking for library z                   : yes 
Checking for library cmocka              : not found 
Checking for function unsetenv           : yes 
Checking for function mkdtemp            : yes 
Checking for function dirname            : yes 
Checking for function basename           : yes 
Checking for function strndup            : yes 
Checking for function strnlen            : yes 
Checking for compiler flags -Wl,--as-needed : yes 
Checking for program '/usr/bin/strip'       : /usr/bin/strip 
Checking for program 'strip'                : /usr/bin/strip 
'configure' finished successfully (0.878s)
'make_all' finished successfully (0.004s)
Waf: Entering directory `/gridsync/build/pyinstaller/bootloader/build/debug'
[ 1/12] Compiling src/pyi_pythonlib.c
[ 2/12] Compiling src/pyi_launch.c
[ 3/12] Compiling src/pyi_archive.c
[ 4/12] Compiling src/pyi_python.c
../../src/pyi_archive.c: In function '_pyi_find_cookie_offset':
../../src/pyi_archive.c:421:9: error: 'for' loop initial declarations are only allowed in C99 mode
         for (size_t i = chunk_size - sizeof(MAGIC) + 1; i > 0; i--) {
         ^
../../src/pyi_archive.c:421:9: note: use option -std=c99 or -std=gnu99 to compile your code

Build failed
Waf: Leaving directory `/gridsync/build/pyinstaller/bootloader/build/debug'
 -> task in 'OBJECTS' failed with exit status 1 (run with -v to display more information)
```

As per the error message above, setting `CFLAGS="-std=gnu99"` to the make/build environment may be sufficient to compile the bootloader. Some additional testing should be performed across other supported operating systems in order to ensure that no other regressions are present before upgrading, however.

--- a/Makefile
+++ b/Makefile
@@ -198,7 +198,7 @@ pyinstaller:
 	rm -rf build/pyinstaller ; \
 	git clone https://github.com/pyinstaller/pyinstaller.git build/pyinstaller && \
 	pushd build/pyinstaller && \
-	git checkout --force v4.2 && \
+	git checkout --force v4.3 && \
 	pushd bootloader && \
 	case `uname` in \
 		Darwin) \

--- a/Makefile
+++ b/Makefile
@@ -211,6 +211,8 @@ pyinstaller:
 		*) \
 			if [ $$(python -c "import distro;print(distro.id() + distro.version())") != "centos7" ] ; then \
 				export CC="gcc -no-pie" ; \
+			else \
+				export CFLAGS="-std=gnu99" ; \
 			fi \
 		;; \
 		esac && \

--- a/Makefile
+++ b/Makefile
@@ -209,10 +209,10 @@ pyinstaller:
 			export LINKFLAGS=-mmacosx-version-min=10.13 \
 		;; \
 		*) \
-			if [ $$(python -c "import distro;print(distro.id() + distro.version())") != "centos7" ] ; then \
-				export CC="gcc -no-pie" ; \
-			else \
+			if [ $$(python -c "import distro;print(distro.id() + distro.version())") == "centos7" ] ; then \
 				export CFLAGS="-std=gnu99" ; \
+			else \
+				export CC="gcc -no-pie" ; \
 			fi \
 		;; \
 		esac && \

--- a/requirements/pyinstaller.txt
+++ b/requirements/pyinstaller.txt
@@ -14,8 +14,8 @@ pefile==2019.4.18; sys_platform == "win32" \
     --hash=sha256:a5d6e8305c6b210849b47a6174ddf9c452b2888340b8177874b862ba6c207645
 PyInstaller==3.6 ; python_version < "3" \
     --hash=sha256:3730fa80d088f8bb7084d32480eb87cbb4ddb64123363763cf8f2a1378c1c4b7
-pyinstaller==4.2; python_version >= "3" \
-    --hash=sha256:f5c0eeb2aa663cce9a5404292c0195011fa500a6501c873a466b2e8cad3c950c
+pyinstaller==4.3; python_version >= "3" \
+    --hash=sha256:5ecf8bbc230d7298a796e52bb745b95eee12878d141f1645612c99246ecd23f2
 pyinstaller-hooks-contrib==2021.1 \
     --hash=sha256:27558072021857d89524c42136feaa2ffe4f003f1bdf0278f9b24f6902c1759c \
     --hash=sha256:892310e6363655838485ee748bf1c5e5cade7963686d9af8650ee218a3e0b031

